priiloader/hacks_hash.ini

```diff
@@ -1,3 +1,115 @@

```

Comment:
correct me if im wrong,
but how different is the patch/code in every system menu version? 
i saw a few instruction differences but thats it (probably memory address differences between versions?)
is there a way to make these into a 1 patch for all?
or could we make this into a 'base' patch & have a version specific patch that changes those few instructions?

also, is it possible to change this in a few patch lines? its just so we have a patch that has this and we can test that when we need to.

Reply:
That would require the user to enable one "generic" hack, and then one hack for their system version, no? Also, I'm generating these patch lines straight from a compiled binary (and I'd like to keep it that way so I can make changes easily if needed), I don't really think manipulating the finished binary is a good idea. 

The only changes in there indeed are just a couple function calls to functions within the system menu, that are at different RAM adresses in each version.

Also, what do you mean with "change this in a few patch lines"? You mean testing if having multiple patch lines for one offset works? That does work, I already tested that.

Reply:
yes it would require one generic hack, and one system version hack. would reduce file size but trades in user convenience. 
i don't know how much larger the hacks file would become though with all those hacks. the binary isn't that tiny and for just 3 - 4 changes i wondered if making it more generic wasn't a good idea. 

i understand that in your development environment its hard to generate though. youre compiling a binary per system menu version?

ye it was for testing that. i know it works but in the future if changes are required the general hacks file in the repo will help to test it.
but np if your methods of generating the hacks doesn't easily allow it for. i can make a test hack to test it if i would ever need to again

Reply:
Yeah, the file size of the hacks_hash.ini would increase a bit, but is that a problem? It still is "just" 30KB in total which isn't that much. 

Yeah, I have my ASM source file and depending on the system menu version I'm including a different include file that has lines like `.set patch_return_offset, 0x8137c094`, then I can just use the names in the source code without having to worry about different versions. Then to create the hacks_hash entries I just compiled it a bunch of times, once for each include file.

Reply:
ah ye, figured as much you were using some asm stuff to generate a binary and set the address like that. 
anyway, we'll see if it causes issues. for now, lets keep it as is.

Reply:
Maybe, if you do want it to become smaller, you could add a feature to Priiloader that allows me to set a parameter for one specific hack, that makes Priiloader automatically enable another hack?

Like I could make one generic hack with all the code that is marked as hidden (and not displayed to the user), and then there are a bunch of small hacks that force-enable that hidden hack and then patch its contents?